src/filesystem/unix/SDL_sysfilesystem.c

```diff
@@ -332,4 +333,122 @@ SDL_GetPrefPath(const char *org, const char *app)
     return retval;
 }
 
+char *
+getpath_try_xdg(SDL_Folder folder)
+{
+    const char *param = NULL;
+    char cmd[32]; /* 32 bytes should be enough, increase if needed */
+    FILE *xdg_proc;
+    char buffer[4096]; /* Apparently PATH_MAX is unreliable */
+    size_t bytes_read;
+    char *retval;
+
+    /* According to `man xdg-user-dir`, the possible values are:
+        DESKTOP
+        DOWNLOAD
+        TEMPLATES
+        PUBLICSHARE
+        DOCUMENTS
+        MUSIC
+        PICTURES
+        VIDEOS
+    */
+    switch(folder)
+    {
+        case SDL_FOLDER_HOME:
+            return SDL_getenv("HOME");
```

Comment:
In pathological situations this can be NULL, in which case this function should probably set the SDL error indicator?

Reply:
Thanks for the catch - not only this, but I believe strings returned by getenv are not free()-able, so I should've duplicated the string to user-owned memory before doing this. My temptation for tiny code made me forget my responsibilities!